### PR TITLE
feat(emqx_bridge): Refactor bridges api

### PR DESCRIPTION
**Pull request by olcai**
This PR refactors the `/bridges` API to make it more consistent with other APIs:
- bridge enable/disable is now done via the endpoint PUT `/bridges/{id}/enable/[true,false]`
- `/bridges/{id}/operation/{operation}` endpoints are now `/bridges/{id}/{operation}`
- metrics are moved out from the GET `/bridges/{id}` response and can now be fetched via `/bridges/{id}/metrics`
- the `bridges/{id}/reset_metrics` endpoint is now `/bridges/{id}/metrics/reset`

Fixes EMQX-7866.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [X] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [X] Change log has been added to `changes/<version>/(feat|fix)-<PR-id>.en.md` and `.zh.md` files
- [X] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

### CODE REVIEW

File: apps/emqx_bridge/i18n/emqx_bridge_api.conf [hunk 1]
```diff
@@ -2,8 +2,8 @@ emqx_bridge_api {
 
     desc_param_path_operation_cluster {
                    desc {
-                         en: """Operations can be one of: enable, disable, start, stop, restart"""
-                         zh: """集群可用操作：启用、禁用、启动、停止、重新启动"""
+                         en: """Operations can be one of: start, stop, restart"""
+                         zh: """"""
```
Comment (lafirest) [inline apps/emqx_bridge/i18n/emqx_bridge_api.conf:6]: zh: """集群可用操作：启用、启动、重新启动"""

File: apps/emqx_bridge/i18n/emqx_bridge_api.conf [hunk 2]
```diff
@@ -44,6 +44,16 @@ emqx_bridge_api {
...
                   }
 
+    desc_param_path_enable {
+                   desc {
+                         en: """Whether or not the bridge is enabled"""
+                         zh: """"""
+                        }
+                   label: {
+                           en: "Enable bridge"
+                           zh: ""
```
Comment (lafirest) [inline apps/emqx_bridge/i18n/emqx_bridge_api.conf:54]: 启用桥接

File: apps/emqx_bridge/i18n/emqx_bridge_api.conf [hunk 3]
```diff
@@ -44,6 +44,16 @@ emqx_bridge_api {
                           }
                   }
 
+    desc_param_path_enable {
+                   desc {
+                         en: """Whether or not the bridge is enabled"""
+                         zh: """"""
```
Comment (lafirest) [inline apps/emqx_bridge/i18n/emqx_bridge_api.conf:50]: 是否启用桥接

File: apps/emqx_bridge/i18n/emqx_bridge_api.conf [hunk 4]
```diff
@@ -112,8 +122,8 @@ emqx_bridge_api {
 
     desc_api7 {
                    desc {
-                         en: """Enable/Disable/Stop/Restart bridges on all nodes in the cluster."""
-                         zh: """在集群中的所有节点上启用/禁用/停止/重新启动 Bridge。"""
+                         en: """Stop/Restart bridges on all nodes in the cluster."""
+                         zh: """"""
```
Comment (lafirest) [inline apps/emqx_bridge/i18n/emqx_bridge_api.conf:126]: 停止或启用所有节点上的桥接

File: apps/emqx_bridge/i18n/emqx_bridge_api.conf [hunk 5]
```diff
@@ -123,15 +133,34 @@ emqx_bridge_api {
...
                   }
 
+    desc_bridge_metrics {
+                   desc {
+                         en: """Get bridge metrics by Id"""
+                         zh: """"""
+                        }
+                   label: {
+                           en: "Get Bridge Metrics"
+                           zh: ""
```
Comment (lafirest) [inline apps/emqx_bridge/i18n/emqx_bridge_api.conf:152]: 获取桥接的指标

File: apps/emqx_bridge/i18n/emqx_bridge_api.conf [hunk 6]
```diff
@@ -123,15 +133,34 @@ emqx_bridge_api {
...
+                          }
+                  }
+
+    desc_enable_bridge {
+                   desc {
+                         en: """Enable or Disable bridges on all nodes in the cluster."""
+                         zh: """"""
```
Comment (lafirest) [inline apps/emqx_bridge/i18n/emqx_bridge_api.conf:159]: 启用或禁用所有节点上的桥接

File: apps/emqx_bridge/i18n/emqx_bridge_api.conf [hunk 7]
```diff
@@ -123,15 +133,34 @@ emqx_bridge_api {
...
+                        }
+                   label: {
+                           en: "Cluster Bridge Enable"
+                           zh: ""
```
Comment (lafirest) [inline apps/emqx_bridge/i18n/emqx_bridge_api.conf:163]: 是否启用集群内的桥接

File: changes/v5.0.15/feat-9736.zh.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+Refactor of /bridges API to make it more consistent with other APIs:
+- bridge enable/disable is now done via the endpoint `/bridges/{id}/enable/[true,false]`
+- `/bridges/{id}/operation/{operation}` endpoints are now `/bridges/{id}/{operation}`
+- metrics are moved out from the GET `/bridges/{id}` response and can now be fetched via `/bridges/{id}/metrics`
+- the `bridges/{id}/reset_metrics` endpoint is now `/bridges/{id}/metrics/reset`
```
Comment (olcai) [inline changes/v5.0.15/feat-9736.zh.md:5]: This would also need translation, please.

**Review by sstrigler — APPROVED** [review]

File: changes/v5.0.15/feat-9736.zh.md (hunk 8 above)
Comment (olcai) [reply changes/v5.0.15/feat-9736.zh.md:5]: @lafirest Could you take a stab at the changes too?

Comment (lafirest) [reply changes/v5.0.15/feat-9736.zh.md:5]: 重构部分 /bridges 的API 使得其和其他 API 能够更加一致：
- 桥接的启用和禁用现在是通过 `/bridges/{id}/enable/[true,false]` API 来实现的
- 使用 `/bridges/{id}/{operation}` 替换了旧的 `/bridges/{id}/operation/{operation}` API
- 指标数据从 `/bridges/{id}` 的响应消息中移除，现在可以使用新的 API  `/bridges/{id}/metrics` 进行访问
- 使用  `/bridges/{id}/metrics/reset` 替换了旧的 `bridges/{id}/reset_metrics` API

File: apps/emqx_bridge/src/emqx_bridge_api.erl [hunk 9]
```diff
@@ -506,14 +563,26 @@ lookup_from_local_node(BridgeType, BridgeName) ->
...
+
+'/bridges/:id/:operation'(post, #{
+    bindings :=
+        #{id := Id, operation := Op}
+}) ->
+    ?TRY_PARSE_ID(
+        Id,
+        case operation_func(Op) of
+            invalid ->
+                {400, error_msg('BAD_REQUEST', <<"invalid operation">>)};
```
Comment (thalesmg) [inline apps/emqx_bridge/src/emqx_bridge_api.erl:578]: Q: isn't it also possible that the bridge does not exist on all nodes?

File: apps/emqx_bridge/test/emqx_bridge_api_SUITE.erl [hunk 10]
```diff
@@ -599,10 +587,95 @@ t_with_redact_update(_Config) ->
...
+    ?assertMatch(
+        #{
+            <<"metrics">> := #{<<"success">> := 1},
+            <<"node_metrics">> := [_ | _]
+        },
+        jsx:decode(Bridge2Str)
+    ),
+
+    %% check for non-empty metrics when listing all bridges
+    {ok, 200, BridgesStr} = request(get, uri(["bridges"]), []),
```
Comment (thalesmg) [inline apps/emqx_bridge/test/emqx_bridge_api_SUITE.erl:659]: Should we also assert that `GET /bridges/:id` does not contain the metrics keys anymore?

File: changes/v5.0.15/feat-9736.zh.md (hunk 8 above)
Comment (olcai) [reply changes/v5.0.15/feat-9736.zh.md:5]: Thanks a ton!

File: apps/emqx_bridge/src/emqx_bridge_api.erl [hunk 11]
```diff
@@ -87,7 +91,7 @@ get_response_body_schema() ->
 param_path_operation_cluster() ->
     {operation,
         mk(
-            enum([enable, disable, stop, restart]),
+            enum([stop, restart]),
```
[the remark this answers is not in the document]
Comment (olcai) [reply apps/emqx_bridge/src/emqx_bridge_api.erl:94]: Ah, good find. The only available ones are stop and restart in the implementation. Will fix the docs and example.

File: apps/emqx_bridge/src/emqx_bridge_api.erl [hunk 12]
```diff
@@ -132,19 +136,33 @@ param_path_id() ->
             }
         )}.
 
-bridge_info_array_example(Method) ->
-    [Config || #{value := Config} <- maps:values(bridge_info_examples(Method))].
+param_path_enable() ->
+    {enable,
```
[the remark this answers is not in the document]
Comment (olcai) [reply apps/emqx_bridge/src/emqx_bridge_api.erl:140]: Yes, why not. I took the implementation from `/bridges` but it should be required, yes.

File: apps/emqx_bridge/src/emqx_bridge_api.erl (hunk 9 above)
Comment (olcai) [reply apps/emqx_bridge/src/emqx_bridge_api.erl:578]: Yes, I suppose so. We will return 500 on general RPC errors or 503 on RPC timeouts. Is there something specific you think we should handle? This part is functionally unchanged in the PR.

Comment (thalesmg) [reply apps/emqx_bridge/src/emqx_bridge_api.erl:579]: I think that it was possible before that a [404 was returned by the operation](https://github.com/emqx/emqx/pull/9736/files#diff-9f021bc96805bb93b81edbbaa5c2b651147245181157ad75dba999129c1320f3L502).  Other than that, I don't recall any at the moment.  :thinking:

**Review by thalesmg — DISMISSED** [review]

File: apps/emqx_bridge/src/emqx_bridge_api.erl (hunk 9 above)
Comment (olcai) [reply apps/emqx_bridge/src/emqx_bridge_api.erl:579]: That is true. But the 404 could only be returned by the enable/disable operation, and not by stop/restart. So the behavior is still consistent with the old API.

**Review by thalesmg — APPROVED** [review]

**Comment by olcai** [comment]
Closed due to merging it in https://github.com/emqx/emqx/pull/9706 instead.

**Pull request closed without merging**